libnetwork/netavark/config.go

```diff
@@ -19,6 +20,67 @@ import (
 	"github.com/containers/storage/pkg/stringid"
 )
 
+func sliceContains(s []string, e string) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func sliceRemoveDuplicates(strList []string) []string {
+	list := []string{}
+	for _, item := range strList {
+		fmt.Println(item)
+		if !sliceContains(list, item) {
+			list = append(list, item)
+		}
+	}
+	return list
+}
+
+func (n *netavarkNetwork) NetworkUpdate(name string, options types.NetworkUpdateOptions) error {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+	err := n.loadNetworks()
+	if err != nil {
+		return err
+	}
+	network, err := n.getNetwork(name)
+	if err != nil {
+		return err
+	}
+	networkDNSServersBefore := network.NetworkDNSServers
+	networkDNSServersAfter := []string{}
+	for _, server := range networkDNSServersBefore {
+		if sliceContains(options.RemoveDNSServers, server) {
+			continue
+		}
+		networkDNSServersAfter = append(networkDNSServersAfter, server)
+	}
+	networkDNSServersAfter = append(networkDNSServersAfter, options.AddDNSServers...)
+	networkDNSServersAfter = sliceRemoveDuplicates(networkDNSServersAfter)
+	network.NetworkDNSServers = networkDNSServersAfter
+	newNetwork := network
+	if !reflect.DeepEqual(networkDNSServersBefore, networkDNSServersAfter) {
```

Comment:
in general I find it easier to read with early returns
```
if equal
    return
write config
```

Also AFAIK reflect is slow, I think we should just compare it in a loop. Or even better you do not even need to compare again if you just store above iy you changed it in a bool

Reply:
made it as per the flow which you like. I did not understand the `bool` logic completely since we also have to care of scenario that we are not adding and removing same element, comparing slices looks easier to me but yes I agree reflect might be slow but I think usually working on such small input it would not matter to us but sure we can optimize this later if we see any need.

Reply:
Feel free to unresolve this comment if you think we should optimize it now.